controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -138,35 +162,36 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 	// Check if Status is empty and status annotation is present
 	// Manually restore data.
 	if !r.hostHasStatus(host) {
-		reqLogger.Info("Fetching Status from Annotation")
+		reqLogger.Info("Reconstructing Status from hardwareData and annotation")
 		objStatus, err := r.getHostStatusFromAnnotation(host)
-		if err == nil && objStatus != nil {
+		if err == nil && objStatus != nil && hardwareData.Name != "" {
+			host.Status.HardwareDetails = hardwareData.Spec.HardwareDetails
+			host.Status = *objStatus
+		} else if err == nil && objStatus != nil {
 			host.Status = *objStatus
-			if host.Status.LastUpdated.IsZero() {
-				// Ensure the LastUpdated timestamp in set to avoid
-				// infinite loops if the annotation only contained
-				// part of the status information.
-				t := metav1.Now()
-				host.Status.LastUpdated = &t
-			}
-			errStatus := r.Status().Update(ctx, host)
-			if errStatus != nil {
-				return ctrl.Result{}, errors.Wrap(errStatus, "Could not restore status from annotation")
-			}
-			return ctrl.Result{Requeue: true}, nil
 		}
-		reqLogger.Info("No status cache found")
```

Comment:
It looks like you removed this line of log output, do we no longer need this log info? Or you forget to copy this line?

Reply:
nit: maybe this log trace should be moved

Reply:
@andfasano do you mean after ?

```
		if errStatus != nil {
			return ctrl.Result{}, errors.Wrap(errStatus, "Could not restore status from annotation")
		}
```

Reply:
Updated, it should be relatively close to the previous state. PTAL @andfasano